### ISSUE TITLE
script setup not emit error when template use un import component

### ISSUE THREAD

**Issue by Valar103769** [vuejs/eslint-plugin-vue#2025]
<img width="825" alt="image" src="https://user-images.githubusercontent.com/36926073/199421561-4d2333d5-22b6-4c1a-8704-a8e231dafcee.png">
<img width="840" alt="image" src="https://user-images.githubusercontent.com/36926073/199421645-bc40c961-4bb2-4822-9426-fbaab3a9ae8a.png">

has a  ts error ,  but i can't find the answer

this is my eslintrc

```js
const { Model } = require('echarts')

Model.exports = {
  env: {
    browser: true,
    es2021: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:vue/vue3-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:prettier/recommended',
  ],
  overrides: [],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: '@typescript-eslint/parser',
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
  plugins: ['vue', '@typescript-eslint'],
  rules: {
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
  globals: {
    defineProps: 'readonly',
    defineEmits: 'readonly',
    defineExpose: 'readonly',
    withDefaults: 'readonly',
  },
}

```

**Comment by ota-meshi** [comment]
Please post an issue according to the template. You should also share a reproducible repository. Otherwise I will close the issue.

https://github.com/vuejs/eslint-plugin-vue/issues/new?assignees=&labels=&template=bug_report.md
